cmake/modules/autopas_CodeCoverage.cmake

```
@@ -1,19 +1,73 @@
-option(AUTOPAS_CODE_COVERAGE "Code Coverage" OFF)
-
-if (AUTOPAS_CODE_COVERAGE MATCHES ON)
-    if (CMAKE_BUILD_TYPE MATCHES Debug)
-        message(STATUS "ENABLING CODE COVERAGE TARGET")
-        include(CodeCoverage)
-        set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_COVERAGE} ${CMAKE_CXX_FLAGS_DEBUG}")
-        setup_target_for_coverage(${PROJECT_NAME}_coverage runTests coverage)
-        setup_target_for_coverage_cobertura(
-            ${PROJECT_NAME}_cobertura
-            runTests
-            coverage
-            --gtest_output=xml:coverage.junit.xml
-            .*gtest.*
-        )
-    else ()
-        message(FATAL_ERROR "Code Coverage only works with CMAKE_BUILD_TYPE = Debug")
-    endif ()
-endif ()
+set(LCOV_ENABLED_MSG "LCOV Coverage report enabled")
+set(LCOV_DISABLED_MSG "LCOV Coverage report disabled")
+set(LCOV_REQUIRED_VERSION_MSG "Disabling code coverage. LCOV version 2.0 or higher is required.")
+set(DISABLE_COVERAGE_MSG "Disabling code coverage. CMAKE_BUILD_TYPE=Debug and AUTOPAS_BUILD_TESTS=ON required.")
+set(COVERAGE_REQUIREMENTS_MSG "Enable code coverage, requires GCC and CMAKE_BUILD_TYPE=Debug.")
+set(NON_GNU_COMPILER_ERROR_MSG "For code coverage ensure that GCOV, LCOV and GENHTML is installed. Disabling code coverage.")
+
+option(AUTOPAS_ENABLE_COVERAGE ${COVERAGE_REQUIREMENTS_MSG} OFF)
+
+if (AUTOPAS_ENABLE_COVERAGE)
+
+  # search for required tools (lcov >= 2.1 is required)
+  find_program(GCOV_EXECUTABLE gcov)
+  find_program(LCOV_EXECUTABLE lcov)
+  find_program(GENHTML_EXECUTABLE genhtml)
+
+  if (NOT GCOV_EXECUTABLE OR NOT LCOV_EXECUTABLE OR NOT GENHTML_EXECUTABLE)
+    message(FATAL_ERROR ${COVERAGE_REQUIREMENTS_MSG})
+    else()
+    message(STATUS "GCOV found at: "${GCOV_EXECUTABLE})
+    message(STATUS "LCOV found at: "${LCOV_EXECUTABLE})
+    message(STATUS "GENHATML found at: "${GENHTML_EXECUTABLE})
```

```suggestion
    message(STATUS "GCOV found at: ${GCOV_EXECUTABLE}")
    message(STATUS "LCOV found at: ${LCOV_EXECUTABLE}")
    message(STATUS "GENHATML found at: ${GENHTML_EXECUTABLE}")
```

Having no space between the second and third argument throws a warning. Solution: have no third argument.